crates/apollo-router/src/configuration/otlp/mod.rs

```diff
@@ -80,6 +80,37 @@ impl ExportConfig {
     }
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct EndpointUrl {
+    #[serde(deserialize_with = "endpoint_url")]
```

Comment:
You can probably do this trick in the ExportConfig struct directly and skip EndpointUrl struct entirely

Reply:
I tried various approaches with or without EndpointUrl, like `serde(flatten)` and those did not work. Unfortunately I do not remember the error messages

Reply:
```rust
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(deny_unknown_fields, rename_all = "snake_case")]
pub struct ExportConfig {
    #[serde(deserialize_with = "endpoint_url")]
    pub endpoint: Option<Url>,
    pub protocol: Option<Protocol>,
    pub timeout: Option<u64>,
}

fn endpoint_url<'de, D>(deserializer: D) -> Result<Option<Url>, D::Error>
where
    D: serde::Deserializer<'de>,
{
    let buf = String::deserialize(deserializer)?;
    Ok(Some(Url::parse(&buf).map_err(serde::de::Error::custom)?))
}
```

Reply:
With the http fix:
```rust
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(deny_unknown_fields, rename_all = "snake_case")]
pub struct ExportConfig {
    #[serde(deserialize_with = "endpoint_url")]
    pub endpoint: Option<Url>,
    pub protocol: Option<Protocol>,
    pub timeout: Option<u64>,
}

fn endpoint_url<'de, D>(deserializer: D) -> Result<Option<Url>, D::Error>
where
    D: serde::Deserializer<'de>,
{
    let mut buf = String::deserialize(deserializer)?;
    if !buf.starts_with("http") {
        buf.insert_str(0, "https://");
    }
    Ok(Some(Url::parse(&buf).map_err(serde::de::Error::custom)?))
}
```

Reply:
did you test it? That was the first thing I tried. IIRC It complains that it expects to find an endpoint field.
From what I understand, with that annotation, serde-json first looks for an endpoint field, then passes its content to the function that then returns an Option. Instead of recognizing the field as optional

Reply:
ah yes you need to add `default` so it can fallback to None before it fails to parse:
```rust
    #[serde(deserialize_with = "endpoint_url", default)]
```

Reply:
nice! updating the code rn